lambda/trmc.ml

```diff
@@ -0,0 +1,859 @@
+               in the [direct] and [dps] versions. *)
+            false;
+          direct = (fun () ->
+            if not settable.benefits_from_dps then
+              k flag (plug_args before (settable.direct ()) after)
+            else
+              let_block_in @@ fun block_var ->
+              Lsequence(Dps.run settable.dps ~tail:false ~dst:(block_dst block_var),
+                        Lvar block_var)
+          );
+          dps = Linear_static (fun ~tail ~dst ~delayed ->
+            match settable.dps with
+            | Dynamic dps ->
+                let_block_in @@ fun block_var ->
+                Lsequence(assign_to_dst loc dst (delayed (Lvar block_var)),
+                          dps ~tail ~dst:(block_dst block_var))
+            | Linear_static dps ->
+                (* We are going to delay the application of the
+                   constructor to the place where our linear-static
+                   argument sets its destination. This may move the
+                   constructor application below some effectful
+                   expressions (if our subterm is of the form [foo;
+                   bar_with_trmc_inside] for example), and we want to
+                   preserve the evaluation order of the other
+                   arguments of the constructor. So we bind them here,
+                   unless are obviously side-effect-frees. *)
+                let bind_list name lambdas k =
+                  let can_be_delayed =
+                    (* Note that the delayed subterms will be used
+                       exactly once in the linear-static subterm. So
+                       we are happy to delay constants, which we would
+                       not want to duplicate. *)
+                    function
+                    | Lvar _ | Lconst _ -> true
+                    | _ -> false in
+                  let bindings, args =
+                    lambdas
+                    |> List.mapi (fun i lam ->
+                      if can_be_delayed lam then (None, lam)
+                      else begin
+                        let v = Ident.create_local (Printf.sprintf "arg_%s_%d" name i) in
+                        (Some (v, lam), Lvar v)
+                      end)
+                    |> List.split in
+                  let body = k args in
+                  List.fold_right (fun binding body ->
+                    match binding with
+                    | None -> body
+                    | Some (v, lam) -> Llet(Strict, Pgenval, v, lam, body)
+                  ) bindings body
+                in
+                bind_list "before" before @@ fun vbefore ->
+                bind_list "after" after @@ fun vafter ->
+                dps ~tail ~dst ~delayed:(fun t ->
+                  delayed (k flag (plug_args vbefore t vafter)))
+          );
+        }
+
+  and choice_prim ctx ~tail prim primargs loc =
+    begin [@warning "-8"] (* see choice *)
```

Comment:
This annotation makes us miss the following primitives:

`| Pignore | Prevapply | Pdirapply | Pcompare_ints | Pcompare_floats | Pcompare_bints _`